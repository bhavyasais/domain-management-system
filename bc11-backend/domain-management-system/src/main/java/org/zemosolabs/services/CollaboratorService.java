package org.zemosolabs.services;

import org.zemosolabs.dto.CollaboratorDTO;

import java.util.List;

public interface CollaboratorService {
    List<CollaboratorDTO> getCollaboratos();
}
