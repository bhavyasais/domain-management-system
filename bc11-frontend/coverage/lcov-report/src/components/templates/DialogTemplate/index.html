
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/components/templates/DialogTemplate</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/components/templates/DialogTemplate</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">82.43% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>61/74</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">50% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>1/2</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.62% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>22/26</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">83.1% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>59/71</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="AddNewDeptTemplate.js"><a href="AddNewDeptTemplate.js.html">AddNewDeptTemplate.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file high" data-value="AddNewRowTemplateDG.js"><a href="AddNewRowTemplateDG.js.html">AddNewRowTemplateDG.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	</tr>

<tr>
	<td class="file high" data-value="AddNewRowTemplateTG.js"><a href="AddNewRowTemplateTG.js.html">AddNewRowTemplateTG.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="13" class="abs high">13/13</td>
	</tr>

<tr>
	<td class="file high" data-value="AddNewTabTemplate.js"><a href="AddNewTabTemplate.js.html">AddNewTabTemplate.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="7" class="abs high">7/7</td>
	</tr>

<tr>
	<td class="file medium" data-value="CollabCardsDialogTemplate.js"><a href="CollabCardsDialogTemplate.js.html">CollabCardsDialogTemplate.js</a></td>
	<td data-value="61.76" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 61%"></div><div class="cover-empty" style="width: 39%"></div></div>
	</td>
	<td data-value="61.76" class="pct medium">61.76%</td>
	<td data-value="34" class="abs medium">21/34</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="2" class="abs medium">1/2</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="61.29" class="pct medium">61.29%</td>
	<td data-value="31" class="abs medium">19/31</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Jan 04 2021 10:42:58 GMT+0530 (India Standard Time)
            </div>
        </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    