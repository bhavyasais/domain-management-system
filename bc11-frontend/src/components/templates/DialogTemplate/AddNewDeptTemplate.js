import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { Typography } from "@material-ui/core";

export default function AddNewDeptTemplate(props) {

    const handleAddClick = () => {
        props.onAddClick(department);
        props.handleClose();
    }

    const [department,setDepartment]= useState('');
    const handleDepartmentChange = (e) => {
        setDepartment(e.target.value);
    }


  return (
    <div>
      <Dialog
        open={props.open}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title"><Typography variant="h6">Add New Department</Typography></DialogTitle>
        <DialogContent>
          <form>
            <div>
              <TextField
                id="outlined-basic"
                label="Department"
                variant="outlined"
                onChange={handleDepartmentChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
        <ButtonAtom
          
          variant="outlined"
          value="Close"
          textVariant="caption"
          onClick={props.handleClose}
        />
        <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        
           
        </DialogActions>
      </Dialog>
    </div>
  );
}
