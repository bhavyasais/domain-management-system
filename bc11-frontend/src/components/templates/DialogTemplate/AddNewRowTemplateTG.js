import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { Typography } from "@material-ui/core";

export default function AddNewRowTemplateTG(props) {

    const handleAddClick = () => {
        props.onAddClick({name,trust_score,relationship});
        props.handleClose();
    }

   
    const [name,setDomain]= useState('');
    const [trust_score,setTrustScore]= useState('');
    const [relationship,setRelationship]= useState('');

    
    const handleDomainChange = (e) => {
        setDomain(e.target.value);
    }

    const handleTrustScoreChange = (e) => {
        setTrustScore(e.target.value);
    }

    const handleRelationshipChange = (e) => {
      setRelationship(e.target.value);
    }


  return (
    <div>
      <Dialog
        open={props.open}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title"><Typography variant="h6">Add New Row</Typography></DialogTitle>
        <DialogContent>
          <form>
          
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Domain"
                variant="outlined"
                onChange={handleDomainChange}
              />
            </div>
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Trust Score"
                variant="outlined"
                onChange={handleTrustScoreChange}
              />
            </div>
            <div>
              {" "}
              <TextField
                id="outlined-basic"
                label="Relationship"
                variant="outlined"
                onChange={handleRelationshipChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
        <ButtonAtom
          
          variant="outlined"
          value="Close"
          textVariant="caption"
          onClick={props.handleClose}
        />
        <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        
           
           
        </DialogActions>
      </Dialog>
    </div>
  );
}