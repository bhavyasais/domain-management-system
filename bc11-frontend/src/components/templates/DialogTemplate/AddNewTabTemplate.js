import React, { useState } from 'react';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import ButtonAtom from '../../atoms/Button/ButtonAtom';
import { Typography } from '@material-ui/core';

export default function AddNewTabTemplate(props) {
  const handleAddClick = () => {
    props.onAddClick(tabName);
    props.handleClose();
  };

  const [tabName, setTabName] = useState('');
  const handleTabChange = (e) => {
    setTabName(e.target.value);
  };

  return (
    <div>
      <Dialog open={props.open} onClose={props.handleClose} aria-labelledby="form-dialog-title">
        <Typography variant="h6">Add New Tab</Typography>
        <DialogContent>
          <form>
            <div>
              <TextField
                id="outlined-basic"
                label="Tab Name"
                variant="outlined"
                onChange={handleTabChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
          <ButtonAtom
            variant="outlined"
            value="Close"
            textVariant="caption"
            onClick={props.handleClose}
          />
          <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        </DialogActions>
      </Dialog>
    </div>
  );
}
