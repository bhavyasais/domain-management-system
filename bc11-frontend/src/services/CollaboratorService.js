import API from '../api';

const CollaboratorService = {
  getAllCollaborators: async () => {
    const { data } = await API.get('collaborators');
    return data;
  },
};

export default CollaboratorService;
