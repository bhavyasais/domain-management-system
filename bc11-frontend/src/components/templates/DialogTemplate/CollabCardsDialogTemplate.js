/* eslint-disable import/prefer-default-export */
import React, { useState, useEffect, useCallback } from 'react';
import Grid from '@material-ui/core/Grid';
import { withStyles, makeStyles } from '@material-ui/core/styles';
import Dialog from '@material-ui/core/Dialog';
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import Typography from '@material-ui/core/Typography';
import MuiDialogContent from '@material-ui/core/DialogContent';
import IconButtonAtom from '../../atoms/Icon/IconAtom';
import IconAtom from '../../atoms/Icon/IconAtom';
import options from '../../atoms/shared/ListIcons';
import CollaboratorService from '../../../services/CollaboratorService';
import CollabCard from '../../molecules/CollabCard/CollabCard';
import GoogleSuiteService from '../../../services/GoogleSuiteService';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
    minWidth: '926px',
    width: '926px',
    height: '544px',
  },
});

const styles = (theme) => ({
  root: {
    margin: 0,
    padding: theme.spacing(3),
  },
  closeButton: {
    position: 'absolute',
    right: theme.spacing(1),
    top: theme.spacing(1),
    color: theme.palette.grey[500],
  },
});

const DialogTitle = withStyles(styles)((props) => {
  const { children, classes, onClose, ...other } = props;
  return (
    <MuiDialogTitle disableTypography className={classes.root} {...other}>
      <Typography variant="h6">{children}</Typography>
      {onClose ? (
        <IconButtonAtom aria-label="close" className={classes.closeButton} onClick={onClose}>
          <IconAtom children={options.close} />
        </IconButtonAtom>
      ) : null}
    </MuiDialogTitle>
  );
});

const DialogContent = withStyles((theme) => ({
  root: {
    padding: theme.spacing(2),
  },
}))(MuiDialogContent);

export const DialogTemplate = ({ handleClose, open, onClick }) => {
  const [collaborations, setCollaborations] = useState([]);

  const handleGoogleSignIn = async (item) => {
    try {
      const GoogleAuth = await window.gapi.auth2.getAuthInstance();
      const result = await GoogleAuth.grantOfflineAccess();
      console.log('result after login', result.code);
      const params = new URLSearchParams();
      params.append('authCode', result?.code);
      await GoogleSuiteService.updateToken(params);
      localStorage.setItem('Collaboration', item.id);
      onClick(item);
    } catch (error) {
      console.warn('Error in gogle sign in', error);
    }
  };

  const handleOnClick = (item) => async () => {
    await handleGoogleSignIn(item);
  };
  const classes1 = useStyles();

  const fetchCollaborations = useCallback(async () => {
    try {
      const result = await CollaboratorService.getAllCollaborators();
      setCollaborations(result);
    } catch (error) {
      console.warn('error in fetching departments : ', error);
    }
  }, []);
  useEffect(() => {
    fetchCollaborations();
  }, []);

  return (
    <div className={classes1.root}>
      <Dialog
        onClose={handleClose}
        aria-labelledby="customized-dialog-title"
        open={open}
        maxWidth="md"
      >
        <DialogTitle id="customized-dialog-title" onClose={handleClose}>
          Choose Collaboration Platforms
        </DialogTitle>
        <DialogContent dividers>
          <Grid container spacing={4}>
            {collaborations.map((item, idx) => (
              <Grid item key={idx}>
                <CollabCard
                  title={item.name}
                  image={item.imageURL}
                  component="img"
                  height={item.height}
                  name={item.name}
                  onClick={handleOnClick(item)}
                />
              </Grid>
            ))}
            ,
          </Grid>
        </DialogContent>
      </Dialog>
    </div>
  );
};
