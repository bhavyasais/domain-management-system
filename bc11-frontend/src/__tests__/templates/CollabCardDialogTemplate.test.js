import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { DialogTemplate } from "../../components/templates/DialogTemplate/CollabCardsDialogTemplate";
import CollaboratorService from "../../services/CollaboratorService";

jest.mock("../../services/CollaboratorService");

const COLLABRATORS = [
  {
    id: "a353351e-9397-4b39-a4ef-521391212f0e",
    name: "Box",
    image:
      "https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/components/36eb3f60-ef31-4b44-be5d-766f4ec7b513.png",
  },
  {
    id: "c148171d-e303-4889-9f19-b30b6bc57340",
    name: "Google Suite",
    image:
      "https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/components/39a2bbdb-7ccc-4347-b341-10fed1434693.png",
  },
];

describe("<CollabCardDialogTemplate/>", () => {
  let mounted;
  const handleOnClose = jest.fn();
  const onCLick = jest.fn();

  beforeEach(async () => {
    act(() => {
      mounted = mount(
        <DialogTemplate
          handleClose={handleOnClose}
          onClick={onCLick}
          open={true}
        />
      );
    });
  });

  it("Renders correctly", () => {
    let wrapper;
    wrapper = renderer.create(
      <DialogTemplate handleClose={() => {}} onClick={() => {}} open={false} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it("Service Calls works correctly", async () => {
    CollaboratorService.getAllCollaborators.mockReturnValue(COLLABRATORS);
    await act(async () => {
      await Promise.resolve(mounted);
      mounted.update();
    });
  });
});
