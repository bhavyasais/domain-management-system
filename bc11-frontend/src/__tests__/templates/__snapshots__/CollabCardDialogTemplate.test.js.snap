// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CollabCardDialogTemplate/> Renders correctly 1`] = `
<div
  className="makeStyles-root-5"
/>
`;
